### ISSUE TITLE
`server.close` does not clean up data

### ISSUE THREAD

**Issue by pags** [mhart/dynalite#75]
I'm not sure what expected behavior is, but closing the Dynalite server doesn't actually tear down any data created during the server's life.

I can't find a place where access to the underlying `db` object is provided, so manual cleanup seems to not be an option.

I did trying requiring memdown externally and calling `clearGlobalStore`, but that seems to put things in a weird state.

**Comment by pags** [comment]
Additionally, trying to manually delete\recreate each table doesn't seem to work either - on query of the recreated tables, previously created data is returned.

**Comment by mhart** [comment]
@pags `server.close` isn't intended to delete data, just close the server (especially important when using an on-disk LevelDB store)

If you're doing this in memory, it's probably easiest just to create a new server each time?

**Comment by mhart** [comment]
@pags can you show me what errors you're getting when you delete and recreate tables? There are many tests for this, so that sounds like a bug

**Comment by pags** [comment]
Thanks for the quick reply...I suspected what you're saying about `close` functionality.

Interestingly enough, I tried doing what you suggest - calling `close` on existing server, then creating a new server running on the same port...using `aws-sdk` to query, I still get old data.  I don't think `aws-sdk` includes any sort of cache by default (and certainly haven't had this issue running against a real Dynamo instance).

To be clear, I'm not getting any errors when deleting\recreating tables in the other scenario...when I say they "don't work", I actually mean they appear to work, but then I still get old data when querying.

**Comment by mhart** [comment]
Hmmm, can you show me a snippet of what you're doing? A new store is created each time the server is instantiated: https://github.com/mhart/dynalite/blob/master/index.js#L23

Unless `memdown` is stateful in some way...? https://github.com/mhart/dynalite/blob/master/db/index.js#L48

**Comment by mhart** [comment]
Actually, I think it might be stateful – if `location` is provided (which it is), then the global store is used... https://github.com/Level/memdown/blob/master/memdown.js#L126

I believe I added the `'/does/not/matter'` on this line many moons ago because without it levelup/memdown used to crash – but maybe it's better if I get rid of it so that it doesn't use the global store.

Can you try and see if it solves your use case to just delete that string so that the line now looks like:

```js
 var db = levelup(options.path, options.path ? {} : {db: memdown}),
    // ...
```

All tests pass, so I think it should be ok (unless there's anyone out there relying on that behavior...)

**Comment by pags** [comment]
Changing that lines seems to result in the same "weird state" that I experienced when calling `clearGlobalStore` directly on memdown (I mentioned this in my first comment but didn't elaborate).  What happens is that after stopping the existing server, when I create\start a new server and then recreate each of my tables (and verify their existence by polling with `describeTable`), I get "ResourceNotFoundException: Requested resource not found" when I try to access said tables.

Assuming I'm not doing something completely wrong, it just seems that some bad state is persisting somewhere.

**Comment by pags** [comment]
Scratch previous comment, I _was_ doing something completely wrong.  Your line change seems to have fixed the issue.  Awesome.

**Comment by pags** [comment]
Any chance of that change landing in the near-term?

**Comment by mhart** [comment]
@pags yup, good chance

**Comment by mhart** [comment]
Published as v1.2.0 – thanks for looking into this!

**Comment by pags** [comment]
No problem, thanks for the quick turnaround
